### PR TITLE
fix decision step independence, add names

### PR DESCRIPTION
**Pull request by zmjjmz**
- Fixes https://github.com/titu1994/tf-TabNet/issues/7
- Make TransformBlocks 3&4 and `transform_coef` depend on the decision step
- Add a bunch of names to the layers and recursively summarize the `tabnet`

### CODE REVIEW

**Comment by titu1994** [comment]
Tensorflow 2.0 doesn't really use the name parameter, so is it really needed ? It adds clutter for no particular reason.

**Comment by zmjjmz** [comment]
That's decidedly news to me -- Tensorboard for example still uses the names when grouping together nodes or their weight distributions, e.g.

Screenshot from 2020-06-22 21-17-45

This is fairly helpful for debugging purposes.

**Comment by titu1994** [comment]
Interesting, tensorboard uses it but saved model doesn't. Learned something new today. LGTM. Should I merge ?

**Comment by zmjjmz** [comment]
Yep, all examples pass and all tests run. For reference final validation accuracy for each example (note that none of the seeds are set in the examples, so I wouldn't take small differences at face value).

- `train_iris.py`
  - `master`: 0.9600
  - `issue_7`: 1.0000
- `train_embedding.py`
  - `master`: 0.9980
  - `issue_7`: 0.9760
- `train_mnist.py`
  - `master`: 0.9297
  - `issue_7`: 0.9538
- `train_stacked_mnist.py`
  - `master`: 0.9205
  - `issue_7`: 0.9329

**Comment by titu1994** [comment]
Fantastic ! Really, thanks for the effort !
